Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_net.cpp

```diff
@@ -1057,7 +1057,9 @@ IPCCommandResult CWII_IPC_HLE_Device_net_ip_top::IOCtl(u32 _CommandAddress)
 				}
 
 				Memory::Memset(BufferOut, 0, BufferOutSize);
-				u32 wii_addr = BufferOut + 4 * 3 + 2 * 2;
+
+				// TODO: This is really hacky; is it actually what IOS does?
+				u32 wii_addr = 0x80000000 | (BufferOut + 4 * 3 + 2 * 2);
```

Comment:
IOS generally works with physical addresses, yes.  But wii_addr is a value that gets passed back to the PowerPC side, then dereferenced, so it has to be an effective address, I think.